src/main/java/seedu/address/logic/commands/UnmarkCommand.java

```diff
@@ -38,10 +40,19 @@ public CommandResult execute(Model model) throws CommandException {
 
         StringBuilder result = new StringBuilder();
 
-        for (Index targetIndex : targetIndexes) {
-            if (targetIndex.getZeroBased() >= lastShownList.size()) {
+        boolean unmarkingUnmarkedPerson;
+        for (Index checkIndex : targetIndexes) {
+            if (checkIndex.getZeroBased() >= lastShownList.size()) {
                 throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
             }
+            Person personToCheck = lastShownList.get(checkIndex.getZeroBased());
+            unmarkingUnmarkedPerson = model.checkForMarkedPerson(personToCheck);
```

Comment:
I believe the line should be `unmarkingUnmarkedPerson = model.checkForUnmarkedPerson(personToCheck);`? 🐒